### ISSUE TITLE
Allow API tokens to be active for only apply or only public API endpoints

### ISSUE THREAD

**Issue by frjo** [HyphaApp/hypha#2607]
Or take it even further and allow API tokens to be active for any combinations of API end points.

**Comment by frjo** [comment]
@Parbhat How hard is this to implement? For now we have only one public API call. The one for open rounds, see https://github.com/HyphaApp/hypha/pull/2567.

We would like to create API tokens that only work with the open rounds API call and not give access to everything else for security reasons.

**Comment by Parbhat** [comment]
@frjo Seems to be implementable using the customisation provided by the drf-api-key package. For the example given we can create separate Public and Apply site specific API keys. https://florimondmanca.github.io/djangorestframework-api-key/guide/#api-key-models

**Comment by frjo** [comment]
@sandeepsajan0 I should have commented on this directly yesterday after the metting regarding the new public site that will be using the open rounds API call. I have opted to go with removing all HasAPIKey permissions except for the open rounds API call. Sorry about this!

The PR did show me how to set special permission for a particular function call. I implemented that at https://github.com/HyphaApp/hypha/pull/2638, thanks!

The rest_framework_api_key package actively discourage using the tokens for authentication. The day we need to give external parties access to the apply API I think we need something like "OAuth".
